src/utils.ts

```diff
@@ -0,0 +1,17 @@
+import moment from 'moment';
+import 'moment-timezone';
+
+
+export default function targetTime(targetUrl: string): any {
+  // const url = 'https://timesync.club/?ts=1712745573031&tz=Asia/Tokyo';
+
+  const dataParams = new URL(targetUrl);
+  const timestamp = dataParams.searchParams.get('ts');
+  const timezone = dataParams.searchParams.get('tz');
+
+  let time = moment(Number(timestamp));
+  //@ts-ignore
+  let targetTime = time.tz(timezone).format();
```

Comment:
Maybe let's not format the date to a string here since we're going to need a date not a string outside? You use it like `new Date(targetTime(getClientUrl()))` now but instead could have been just `targetTime(getClientUrl())`. Now it looks a bit weird we return a string from this function just to convert it back to date the very next step.

Reply:
ok,  I do "let targetTime = time.tz(timezone)"